usl_models/usl_models/flood_ml/model.py

```diff
@@ -48,72 +50,102 @@ def _build_model(self) -> tf.keras.Model:
         )
         return model
 
-    def _validate_and_preprocess_inputs(
+    def _validate_and_preprocess_data(
         self,
-        inputs: dict[str, tf.Tensor],
-    ) -> dict[str, tf.Tensor]:
-        """Validates inputs and does all necessary preprocessing.
+        data: FloodModelData,
+        training=True,
+    ) -> FloodModelData:
+        """Validates model data and does all necessary preprocessing.
 
         Args:
-            inputs: A dictionary of input tensors.
-                - "geospatial" (required): [H, W, f]
-                - "temporal" (required): [T]
-                - "spatiotemporal" (optional): a single flood map [H, W]
+            data: A FloodModelData object. Labels are required for training.
+            training: Whether data is used for model training. If True, labels
+                will be validated.
 
         Returns:
-            A dictionary of processed input tensors.
+            A processed FloodModelData object.
         """
-        if "geospatial" not in inputs:
-            raise ValueError("Missing required tensor 'geospatial'.")
-        if "temporal" not in inputs:
-            raise ValueError("Missing required tensor 'temporal'.")
+        if training:
```

Comment:
looks like
```python
if training and data.labels is None:
    raise ValueError("Labels must be provided during model training.")
```

Reply:
Yes. But there was another block to the logic for checking the expected labels. Would you recommend pulling that into its own `if` block?

Reply:
Ah, I missed that! Fine to keep as is. Could also consider
```python
if training and data.labels is None:
    raise ValueError("Labels must be provided during model training.")
elif training:
   ...
```
but I think both are readable (despite me misreading it :)